### PR TITLE
Infinite banner

### PR DESCRIPTION
**Pull request by CristobalMe**

### CODE REVIEW

File: frontend/src/components/Card.jsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
+const Card = ({data}) => {
```
Comment (toriqyanM) [inline frontend/src/components/Card.jsx:1]: This is specifically for event card view right? I would 1) put event in the component name, and 2) instead of using {data}, give props definition {name, location, imageUrl, score, numReviews}. That way you can make sure that the usage of this component would have all the required fields.

File: frontend/src/components/attendingPage/CardGridAttending.jsx [hunk 2]
```diff
@@ -0,0 +1,90 @@
+import Card from "../Card";
+
+
+
+const CardGridAttending = () => {
```
Comment (toriqyanM) [inline frontend/src/components/attendingPage/CardGridAttending.jsx:5]: Is this for events the user has attended or will be attending. Either way, I would put the keyword Event in, i.e. EventToAttendCard

File: frontend/package.json [hunk 3]
```diff
@@ -10,10 +10,14 @@
...
   "dependencies": {
+    "@material-tailwind/react": "^2.1.9",
     "lucide-react": "^0.396.0",
     "react": "^18.3.1",
     "react-dom": "^18.3.1",
-    "react-router-dom": "^6.24.0"
+    "react-router-dom": "^6.24.0",
+    "react-slick": "^0.30.2",
+    "slick-carousel": "^1.8.1",
+    "swiper": "^11.1.4"
```
Comment (toriqyanM) [inline frontend/package.json:20]: Remove this

File: frontend/src/components/favoritesPage/CardGridFavorites.jsx [hunk 4]
```diff
@@ -0,0 +1,90 @@
+import Card from "../Card";
+
+
+
+const CardGridFavorites = () => {
```
Comment (toriqyanM) [inline frontend/src/components/favoritesPage/CardGridFavorites.jsx:5]: Same naming feedback as attending

File: frontend/src/components/header/Header.jsx [hunk 5]
```diff
@@ -34,17 +26,17 @@ const Header = () => {
     };
     
     return (
-      <div>
-        <header className='sticky top-0 z-[1] mx-auto flex w-full max-w-2xl:max-w-7xl flex-wrap items-center justify-between bg-background p-[2em] font-sans font-bold backdrop-blur-[100px] bg-slate-900 text-white'>
-                <div className='mr-10 ml-10'>
-                  <h2 className='font-bebas text-3xl'>Event finder</h2>
+      <div >
```
Comment (toriqyanM) [inline frontend/src/components/header/Header.jsx:29]: unnecessary whitespace

File: frontend/src/App.jsx [hunk 6]
```diff
@@ -12,7 +17,13 @@ function App() {
     <Router>
 
       <Routes>
-        <Route path="/" element={<HomePage />} />
+        <Route path="/" element={<InitialPage />} />
+        <Route path="/SignUp" element={<SignupPage />} />
+        <Route path="/Home" element={<HomePage />} />
```
Comment (toriqyanM) [inline frontend/src/App.jsx:22]: I think some thoughts can be put into user flow here. 
Generally when we land on a website, we don't want to first see a plain login page. We want to see some public content that gives us a sense of what the website offers etc. If we have logged in and cookie hasn't expired, we should have logged in. If not, we can choose to log in or sign up from the nav bar.

File: frontend/src/components/initialPage/InitialHeader.jsx [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+    return (
+      <div >
+        <header className='fixed z-[1] top-0 w-full flex flex-wrap items-center justify-between bg-background p-[2em] font-sans font-bold backdrop-blur-[100px] bg-slate-900 text-white'>
+                <div className='m-3 items-center'>
+                  <h2 className='font-bebas text-xl text-3xl'>Event finder</h2>
+                </div>
+
+                <div className='content-center pl-[2rem]'>
+                  <div className="hidden justify-end lg:flex" id='component'>
+                      <NavLinks />
```
Comment (toriqyanM) [inline frontend/src/components/initialPage/InitialHeader.jsx:23]: Redundant code here with the creation of InitialHeader here. This is largely similar to Header, except it is for when user not logged in. You can achieve this by use a state to tell whether user has logged in or not. If they have, load loggedInNavLinks; if they have not load LoggedOutNavLinks.

File: frontend/src/components/header/Header.jsx [hunk 8]
```diff
@@ -34,17 +26,17 @@ const Header = () => {
...
 
-                <div className='mr-1'>
+                <div className='hidden mr-1 content-center text-black lg:flex'>
                   <SearchBar />
                 </div>
 
-                <div className='flex flex-[1] items-center justify-end flex-wrap items-center'>
+                <div className='content-center'>
                   <div className="hidden justify-end lg:flex" id='component'>
                       <NavLinksLogged />
```
Comment (toriqyanM) [inline frontend/src/components/header/Header.jsx:41]: See comments on your initialHeader. The two header components can be consolidated.

File: frontend/src/components/attendingPage/AttendingPage.jsx [hunk 9]
```diff
@@ -0,0 +1,15 @@
...
+  return (
+    <div>
+
+        <Header />
+        <div className="h-screen flex items-center justify-center lg:mt-[20rem] sm:mt-[45rem]"><CardGridAttending /></div>
+    </div>
+  )
+}
+
+export default App
```
Comment (toriqyanM) [inline frontend/src/components/attendingPage/AttendingPage.jsx:15]: Add event keyword in file and directory name to avoid ambiguity

File: frontend/src/components/carousel/Carousel.jsx [hunk 10]
```diff
@@ -0,0 +1,132 @@
...
+                <CarouselCard data={d} />
+            ))}
+        </Slider>
+
+        
+      </div>
+    );
+  };
+  
+  export default Carousel;
```
Comment (toriqyanM) [inline frontend/src/components/carousel/Carousel.jsx:132]: Same naming feedback

File: frontend/src/components/carousel/Carousel.jsx [hunk 11]
```diff
@@ -0,0 +1,132 @@
+import Slider from "react-slick";
+import "slick-carousel/slick/slick.css";
+import "slick-carousel/slick/slick-theme.css";
+import CarouselCard from "./CarouselCard.jsx";
+
+
+
+const Carousel = () => {
+    const data = [
```
Comment (toriqyanM) [inline frontend/src/components/carousel/Carousel.jsx:9]: Since this test data is repeated everywhere, create a data.js

File: frontend/src/components/signUpPage/InitialHeaderSignUp.jsx [hunk 12]
```diff
@@ -0,0 +1,33 @@
+import { Link } from "react-router-dom"
```
Comment (toriqyanM) [inline frontend/src/components/signUpPage/InitialHeaderSignUp.jsx:1]: This component is exactly the same as InitialHeader.

File: frontend/src/components/signUpPage/SignUpForm.jsx [hunk 13]
```diff
@@ -0,0 +1,63 @@
...
+                            </label>
+                            <input className="border rounded w-full py-2 px-3 text-gray-700 mb-3 leading-tight" id="sex" type="text" placeholder="male/female" />
+                        </div>
+
+                                                
+                        
+
+
+
+
```
Comment (toriqyanM) [inline frontend/src/components/signUpPage/SignUpForm.jsx:52]: Bring back week 1 comment. Remove unnecessary empty lines.

File: frontend/src/components/signUpPage/SignUpForm.jsx [hunk 14]
```diff
@@ -0,0 +1,63 @@
...
+                        <div className="flex items-center justify-between">
+                            <Link to="/Home"><button className="bg-blue-950 hover:bg-blue-700 text-white font-bold py-2 px-4 rounded" type="button">Sign Up</button></Link>
+                            <Link to="/"><button className="text-blue-950 hover:text-blue-700 font-bold py-2 px-4 rounded " type="button">Sign In</button></Link>
```
Comment (toriqyanM) [inline frontend/src/components/signUpPage/SignUpForm.jsx:55]: Are sign in and sign up the same form? With how the design this, user would think they need birthday, sex and confirm password to sign in, but then the navigation takes them to sign in again. Consider the user flow again. You can find a public website and play around with their sign in/up flow.

**Review by toriqyanM — COMMENTED** [review]
Overall the large comment is to iron out the user flow. The flow around sign in and up and initial landing has some hiccups. 
Also don't forget to give user the ability to log out.

**Review by toriqyanM — COMMENTED** [review]
Another general feedback is that pull request should be smaller. It would be much easier to review if it were to be broken into 3 pull requests.

**Review by toriqyanM — CHANGES_REQUESTED** [review]
When you are addressing comments, make sure that you don't include new changes. It makes review really difficult. Remove all your changes on backend. Also squash your commits into one.

File: frontend/package.json [hunk 15]
```diff
@@ -14,10 +14,7 @@
     "lucide-react": "^0.396.0",
     "react": "^18.3.1",
     "react-dom": "^18.3.1",
-    "react-router-dom": "^6.24.0",
-    "react-slick": "^0.30.2",
-    "slick-carousel": "^1.8.1",
```
Comment (toriqyanM) [inline frontend/package.json]: Is this not used in Carousel? Can you verify that this website still loads?

File: frontend/src/App.jsx [hunk 16]
```diff
@@ -1,8 +1,8 @@
 import HomePage from "./components/homePage/HomePage.jsx"
 import ForYouPage from "./components/forYouPage/ForYouPage.jsx"
 import NearbyPage from "./components/nearbyPage/NearbyPage.jsx"
-import AttendingPage from "./components/attendingPage/AttendingPage.jsx"
-import FavoritesPage from "./components/favoritesPage/FavoritesPage.jsx"
+import EventAttendingPage from "./components/eventAttendingPage/EventAttendingPage.jsx"
+import FavoriteEventsPage from "./components/favoriteEventsPage/FavoriteEventsPage.jsx"
```
Comment (toriqyanM) [inline frontend/src/App.jsx:5]: Keep it consistent. Event or Events

File: frontend/package.json (hunk 15 above)
Comment (CristobalMe) [reply frontend/package.json]: Yes, it still loads

Comment (toriqyanM) [reply frontend/package.json]: Can you try running `npm ci` in terminal and see if it loads again? Or your Carousel.jsx is not using slick-carousel module?

Comment (CristobalMe) [reply frontend/package.json]: now it doesn't

Comment (toriqyanM) [reply frontend/package.json]: I realized that I misclicked on my first review. Sorry about that. I was trying to point out that Swiper package is no longer used in your code and should be removed. Essentially `npm ci` uninstall and re-install all npm packages as your package.json describes. You still need those three packages. It was originally loading because they exist in your local node-modules due to previous install.

Comment (CristobalMe) [reply frontend/package.json]: ohh, thank you

**Comment by ygongdev** [comment]
Please remove the `node_modules` folder in this PR. It makes code review very difficult and should be easy to remove, see https://gist.github.com/lmcneel/45594e550a3403d589bdcaad38138a83.

File: frontend/src/components/carousel/CarouselCard.jsx [hunk 17]
```diff
@@ -0,0 +1,33 @@
...
+    return(
+      <div className="max-w-sm rounded overflow-hidden shadow-lg bg-white h-[20rem] w-[15rem]">
+        <img className="h-[10rem] w-[15rem]" src={data.imageUrl} alt="https://www.shutterstock.com/image-vector/default-ui-image-placeholder-wireframes-600nw-1037719192.jpg" />
+        <div className="px-3 py-4 flex flex-col justify-center items-center">
+          <div className="font-bold text-base mb-2 mx-[1.3rem]">{data.name}</div>
+          <p className="text-gray-700 text-sm mx-[1.1rem]">
+            📍{data.location}
+          </p>
+
+
```
Comment (ygongdev) [inline frontend/src/components/carousel/CarouselCard.jsx:11]: nit: reduce unnecessary extra line breaks (shouldn't need more than 1)

File: frontend/src/components/carousel/CarouselCard.jsx [hunk 18]
```diff
@@ -0,0 +1,33 @@
...
+                </svg>
+                <p className="ms-2 text-sm font-bold text-gray-900 dark:text-white">{data.score}</p>
+                <span className="w-1 h-1 mx-1.5 bg-gray-500 rounded-full dark:bg-gray-400"></span>
+                <a className="text-sm font-medium text-gray-900  hover:no-underline dark:text-white">{data.numReviews} reviews</a>
+          </div>
+        </div>
+
+
+        
+
```
Comment (ygongdev) [inline frontend/src/components/carousel/CarouselCard.jsx:28]: nit: same here for extra lines

File: frontend/src/components/homePage/HomePage.jsx [hunk 19]
```diff
@@ -1,11 +1,23 @@
...
-
   return (
     <div>
-      <Header />
+        <Header />
+
+
+        <div className='ml-[7rem] mr-[7rem]  mt-[10rem]'>
+            <h2 className='font-bebas text-white text-xl'>For You</h2>
+            <Carousel />
```
Comment (ygongdev) [inline frontend/src/components/homePage/HomePage.jsx:18]: So when we design an UI application, data flow is important.

There is the concept of [presentational component and container component](https://medium.com/@dan_abramov/smart-and-dumb-components-7ca2f9a7c7d0), which can be useful here.

Ideally, `<Carousel>` should only be responsible for showing UI and handling UI interaction. The data source for it should be coming from outside via `props`, e.g `<Carousel data={someDataExample} />`.

This design will allow you to reuse your carousel component for different data sources as long as it follows the same format.

File: frontend/src/components/homePage/HomePage.jsx [hunk 20]
```diff
@@ -1,11 +1,23 @@
...
+        </div>
+
+        <div className='ml-[7rem] mr-[7rem]  mt-[10rem] content-center'>
+            <CoverFlow />
```
Comment (ygongdev) [inline frontend/src/components/homePage/HomePage.jsx]: Same `<Carousel>` comment here for `<CoverFlow>`

File: backend/.gitignore [hunk 21]
```diff
@@ -1,3 +1,4 @@
 node_modules
+**/node_modules
```
Comment (ygongdev) [inline backend/.gitignore:2]: If possible, for the deleted `node_modules`, let's create a completely separate PR to remove them from the repo, so this PR can focus on the actual code changes.

So the steps should look like:
1. Open new PR to delete the node_modules folder
2. Submit and merge PR into your main branch
3. Rebase these open PRs onto your main branch (should be after the deletion of node_modules from github, but you should still see it in your local repo)

File: frontend/src/App.jsx (hunk 6 above)
Comment (toriqyanM) [reply frontend/src/App.jsx:19]: If you decide to address the user flow comment later, please add TODO in a comment so that it doesn't left forgotten. Add something in code like this:
`// TODO: login/signup routing logic

File: backend/routes/dRoutes.jsx [hunk 22]
```diff
@@ -0,0 +1,42 @@
+const { PrismaClient } = require('@prisma/client')
+const prisma = new PrismaClient()
+
+app.get('/books', async (req, res) => {
```
Comment (yijial) [inline backend/routes/dRoutes.jsx:4]: noob q, why do we use "books" as the directory name? It's supposed to store "routes" if I understand correctly

**Review by yijial — COMMENTED** [review]
Can we breakdown this into several smaller PRs? with each PR only achieve ONE functionality. This way it's easier to understand what's going on and we can provide more detailed feedback if not reviewing 500+ files at a time.

File: frontend/src/components/header/Header.jsx (hunk 5 above)
Comment (CristobalMe) [reply frontend/src/components/header/Header.jsx:29]: Done

File: frontend/src/components/favoritesPage/CardGridFavorites.jsx (hunk 4 above)
Comment (CristobalMe) [reply frontend/src/components/favoritesPage/CardGridFavorites.jsx]: Done

File: frontend/src/components/Card.jsx (hunk 1 above)
Comment (CristobalMe) [reply frontend/src/components/Card.jsx:1]: To do

File: frontend/src/App.jsx (hunk 6 above)
Comment (CristobalMe) [reply frontend/src/App.jsx:19]: To do

File: frontend/src/components/initialPage/InitialHeader.jsx (hunk 7 above)
Comment (CristobalMe) [reply frontend/src/components/initialPage/InitialHeader.jsx:23]: To do (need to solve user auth first)

File: backend/routes/dRoutes.jsx (hunk 22 above)
Comment (CristobalMe) [reply backend/routes/dRoutes.jsx:4]: Done

File: frontend/src/components/homePage/HomePage.jsx (hunk 19 above)
Comment (CristobalMe) [reply frontend/src/components/homePage/HomePage.jsx:18]: Done

File: frontend/src/components/header/Header.jsx (hunk 8 above)
Comment (CristobalMe) [reply frontend/src/components/header/Header.jsx:41]: To do

**Review by toriqyanM — APPROVED** [review]
LGTM

File: backend/.gitignore [hunk 23]
```diff
@@ -1,3 +1,5 @@
 node_modules
+node_modules/
```
[the remark this answers is not in the document]
Comment (CristobalMe) [reply backend/.gitignore]: Done

File: frontend/.gitignore [hunk 24]
```diff
@@ -11,6 +11,8 @@ node_modules
 dist
 dist-ssr
 *.local
+node_modules
+node_modules/
```
[the remark this answers is not in the document]
Comment (CristobalMe) [reply frontend/.gitignore]: Done

File: frontend/src/components/Card.jsx (hunk 1 above)
Comment (CristobalMe) [reply frontend/src/components/Card.jsx]: Done

File: frontend/src/components/attendingPage/CardGridAttending.jsx (hunk 2 above)
Comment (CristobalMe) [reply frontend/src/components/attendingPage/CardGridAttending.jsx]: Done

File: frontend/package.json (hunk 3 above)
Comment (CristobalMe) [reply frontend/package.json]: Done

File: frontend/src/App.jsx (hunk 6 above)
Comment (CristobalMe) [reply frontend/src/App.jsx:23]: Done

File: frontend/src/components/initialPage/InitialHeader.jsx (hunk 7 above)
Comment (CristobalMe) [reply frontend/src/components/initialPage/InitialHeader.jsx:23]: Done

File: frontend/src/components/header/Header.jsx (hunk 8 above)
Comment (CristobalMe) [reply frontend/src/components/header/Header.jsx:38]: Done

File: frontend/src/components/attendingPage/AttendingPage.jsx (hunk 9 above)
Comment (CristobalMe) [reply frontend/src/components/attendingPage/AttendingPage.jsx]: Done

File: frontend/src/components/carousel/Carousel.jsx (hunk 10 above)
Comment (CristobalMe) [reply frontend/src/components/carousel/Carousel.jsx]: Done

File: frontend/src/components/carousel/Carousel.jsx (hunk 11 above)
Comment (CristobalMe) [reply frontend/src/components/carousel/Carousel.jsx]: Done

File: frontend/src/components/signUpPage/InitialHeaderSignUp.jsx (hunk 12 above)
Comment (CristobalMe) [reply frontend/src/components/signUpPage/InitialHeaderSignUp.jsx]: Done

File: frontend/src/components/signUpPage/SignUpForm.jsx (hunk 13 above)
Comment (CristobalMe) [reply frontend/src/components/signUpPage/SignUpForm.jsx]: Done

File: frontend/src/components/signUpPage/SignUpForm.jsx (hunk 14 above)
Comment (CristobalMe) [reply frontend/src/components/signUpPage/SignUpForm.jsx:49]: Done

File: frontend/src/App.jsx (hunk 16 above)
Comment (CristobalMe) [reply frontend/src/App.jsx]: Done

File: frontend/src/components/carousel/CarouselCard.jsx (hunk 17 above)
Comment (CristobalMe) [reply frontend/src/components/carousel/CarouselCard.jsx]: Done

File: frontend/src/components/carousel/CarouselCard.jsx (hunk 18 above)
Comment (CristobalMe) [reply frontend/src/components/carousel/CarouselCard.jsx]: Done

File: frontend/src/components/homePage/HomePage.jsx (hunk 20 above)
Comment (CristobalMe) [reply frontend/src/components/homePage/HomePage.jsx]: Done

File: backend/.gitignore (hunk 21 above)
Comment (CristobalMe) [reply backend/.gitignore]: Done
